packages/twenty-server/src/workspace/messaging/services/gmail-full-sync.service.ts

```diff
@@ -102,27 +115,41 @@ export class GmailFullSyncService {
 
     const messageQueries = createQueriesFromMessageIds(messagesToFetch);
 
+    startTime = Date.now();
+
     const { messages: messagesToSave, errors } =
       await this.fetchMessagesByBatchesService.fetchAllMessages(
         messageQueries,
         accessToken,
+        'gmail full-sync',
+        workspaceId,
+        connectedAccountId,
       );
 
+    endTime = Date.now();
+
+    this.logger.log(
+      `gmail full-sync for workspace ${workspaceId} and account ${connectedAccountId}: fetching all messages in ${
+        endTime - startTime
+      }ms.`,
+    );
+
     if (messagesToSave.length === 0) {
+      if (errors.length) throw new Error('Error fetching messages');
```

Comment:
@bosiraphael should we check if errors is not empty outside of this condition? I think we still push to errors if messagesToSave is not empty right?